src/peerconnection.cpp

```diff
@@ -593,6 +586,51 @@ void PeerConnection::forwardMedia(message_ptr message) {
 			track->incoming(message);
 }
 
+std::optional<std::string> PeerConnection::getMidFromSSRC(SSRC ssrc) {
+    if (auto it = mMidFromSssrc.find(ssrc); it != mMidFromSssrc.end()) {
+        return it->second;
+    } else {
+        std::lock_guard lock(mLocalDescriptionMutex);
+        if (!mLocalDescription)
+            return nullopt;
+
+        for (unsigned int i = 0; i < mRemoteDescription->mediaCount(); ++i) {
+            if (auto found = std::visit(
+                    rtc::overloaded{[&](Description::Application *) -> std::optional<string> {
+                        return std::nullopt;
+                    },
+                                    [&](Description::Media *media) -> std::optional<string> {
+                                        return media->hasSSRC(ssrc)
+                                               ? std::make_optional(media->mid())
+                                               : nullopt;
+                                    }},
+                    mRemoteDescription->media(i))) {
+
+                mMidFromSssrc.emplace(ssrc, *found);
+                return *found;
+            }
+        }
+
+        for (unsigned int i = 0; i < mLocalDescription->mediaCount(); ++i) {
+            if (auto found = std::visit(
+                    rtc::overloaded{[&](Description::Application *) -> std::optional<string> {
+                        return std::nullopt;
+                    },
+                                    [&](Description::Media *media) -> std::optional<string> {
+                                        return media->hasSSRC(ssrc)
+                                               ? std::make_optional(media->mid())
+                                               : nullopt;
+                                    }},
+                    mLocalDescription->media(i))) {
+
+                mMidFromSssrc.emplace(ssrc, *found);
+                return *found;
+            }
+        }
```

Comment:
Just a suggestion: the loop might actually be in a subroutine that you'd call twice with both descriptions, it would prevent duplicating the code.